lib/core/src/chain_swap.rs

```diff
@@ -582,14 +582,28 @@ impl ChainSwapStateHandler {
             lockup_details.amount, lockup_details.lockup_address
         );
 
-        let lockup_tx = self
+        let lockup_tx = match self
             .onchain_wallet
             .build_tx(
                 None,
                 &lockup_details.lockup_address,
                 lockup_details.amount as u64,
             )
-            .await?;
+            .await
+        {
+            Err(PaymentError::InsufficientFunds) => {
```

Comment:
Why do we need the fallback to drain? Why not just return error? As far as I understand the user didn't use explicitly "drain" flag in this flow.

Reply:
At this point (in `pay_onchain` > changed status to `Created` > `on_new_outgoing_status` > `lockup_funds`)

https://github.com/breez/breez-sdk-liquid/blob/b20c27a3998f75bb80f80c3c42352303d774e3df/lib/core/src/chain_swap.rs#L399

we don't know what the user chose. We only know we should lockup `lockup_details.amount` sats.

So the logic iin `lockup_funds` becomes:
- we start by assuming it's a normal payment (non-drain)
- if `InssuficientFunds`, there is a chance this might have been a drain, not a normal payment
  - we try drain
    - if drain succeeds => `Ok`
    - if drain fails => propagate the error

Reply:
I see but there is still a risk here that we should handle.
Does the drain uses the given lockup amount? I am asking because there might be more funds in the wallet were when the swap was created and now we risk in paying lots of fees?
If we use drain here I think we should make sure the amount and fees we pay are as expected.

Reply:
> If we use drain here I think we should make sure the amount and fees we pay are as expected

We do, I added it in 1fa49259d1416138b7ee4f8fb29844dda4d2e8d4 .

The way we build the drain tx takes the expected amount as arg:

https://github.com/breez/breez-sdk-liquid/blob/62751b8448ff2f2a2c147e3bb43b1c709d7dc924/lib/core/src/chain_swap.rs#L594-L603

If LWK builds a drain tx of different amount, for any reason, this `build_drain_tx` will fail with a validation error.

Reply:
So the edge case you brought up is already covered by this drain amount validation.